src/Mobs/Path.cpp

```diff
@@ -193,21 +208,43 @@ bool cPath::Step_Internal()
 	}
 	// process a currentCell by inspecting all neighbors.
 
-	// Check North, South, East, West on all 3 different heights.
-	int i;
-	for (i = -1; i <= 1; ++i)
+
+	// Check North, South, East, West on our height.
```

Comment:
I fixed "diagonal cutting" on all planes, not just XZ. Diagonal cutting is when e.g. there's a block our north and a block our east and we allow ourselves to travel north-east, illegally passing through both the easter and the northern block. The issue was nonexistent with 1x2 mobs, but chickens would have suffered without this fix.